### PR TITLE
Second set of articles by Kristof

### PR DESCRIPTION
**Pull request by TuanaCelik**

### CODE REVIEW

File: content/blog/scaling-nlp-indexing-pipelines-with-keda-and-haystack-part-1/index.md [hunk 1]
```diff
@@ -0,0 +1,437 @@
...
+            if not messages:  
+                return  
+            paths = self.s3_client.download_files(messages)  
+            return paths  
+  
+    def upload_file(self, local_path: Path) -> None:  
+        self.s3_client.upload_file(local_path)  
+        self.sqs_client.publish_key(local_path.name)
+```
+# Set up a local AWS environment with LocalStack
```
Comment (bilgeyucel) [inline content/blog/scaling-nlp-indexing-pipelines-with-keda-and-haystack-part-1/index.md:315]:
```suggestion
## Set up a local AWS environment with LocalStack
```

File: content/blog/scaling-nlp-indexing-pipelines-with-keda-and-haystack-part-1/index.md [hunk 2]
```diff
@@ -0,0 +1,437 @@
...
+haystack-keda-indexing-localstack-1  | 2023-04-22T09:55:48.327  INFO --- [   asgi_gw_0] localstack.request.aws     : AWS sqs.GetQueueUrl => 200
+```
+
+We have a stateless service that processes indexing requests asynchronously and can operate in Docker Compose. We can now deploy this service to Kubernetes, configure KEDA, and scale it.
+
+## Conclusion and next steps
+
+This article explains how to create a stateless consumer application that turns files into documents. We learned that we can use this application with different models, preprocessors, and file types which are defined in a single configuration file using Haystack. In order to ship this application to a production-ready deployment, we will deploy this service on Kubernetes and define custom autoscaling in  [the next chapter](scaling-nlp-indexing-pipelines-with-keda-and-haystack-part-2).
```
Comment (bilgeyucel) [inline content/blog/scaling-nlp-indexing-pipelines-with-keda-and-haystack-part-1/index.md:437]: I think we need to give the full link to the other blog here: https://haystack.deepset.ai/blog/scaling-nlp-indexing-pipelines-with-keda-and-haystack-part-2

File: content/blog/scaling-nlp-indexing-pipelines-with-keda-and-haystack-part-1/index.md [hunk 3]
```diff
@@ -0,0 +1,437 @@
...
+
+But how do we create a service that can turn files into searchable documents? And how can we efficiently scale these applications to index millions of files?
+
+**The goal of this article is to build a system that can run use case-specific indexing pipelines for large-scale production NLP applications.**  It will focus on  **how to create, deploy, and dynamically scale these applications**  using  [Haystack](https://haystack.deepset.ai/)  and  [KEDA](https://keda.sh/). The first part covers creating a Python application with Haystack. The second part details how to deploy and scale the application using  [KEDA](https://keda.sh/)  on  [Kubernetes](https://kubernetes.io/de/).
+
+> This tutorial only covers turning files into text snippets, embeddings or arbitrary preprocessed data. We will not cover the augmented LLM application itself. If you are interested in creating such applications, feel free to use this tutorial:  [https://haystack.deepset.ai/blog/build-a-search-engine-with-gpt-3](https://haystack.deepset.ai/blog/build-a-search-engine-with-gpt-3)
```
Comment (bilgeyucel) [inline content/blog/scaling-nlp-indexing-pipelines-with-keda-and-haystack-part-1/index.md:21]:
```suggestion
> This tutorial only covers turning files into text snippets, embeddings or arbitrary preprocessed data. We will not cover the augmented LLM application itself. If you are interested in creating such applications, feel free to use this tutorial:  [Build a Search Engine with GPT-3](https://haystack.deepset.ai/blog/build-a-search-engine-with-gpt-3)
```

**Review by bilgeyucel — APPROVED** [review]
I left some comments, apart from them LGTM 🚀

File: content/blog/scaling-nlp-indexing-pipelines-with-keda-and-haystack-part-1/index.md (hunk 2 above)
Comment (TuanaCelik) [reply content/blog/scaling-nlp-indexing-pipelines-with-keda-and-haystack-part-1/index.md:437]: no, this is ok :)

**Pull request merged**